core/src/main/scala/com/softwaremill/react/kafka_api_opt3/Example.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.softwaremill.react.kafka_api_opt3
+
+import scala.concurrent.Future
+
+import akka.Done
+import com.softwaremill.react.kafka_api_producer.Producer
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+
+trait Example {
+  implicit val ec = scala.concurrent.ExecutionContext.Implicits.global
+
+  def consumerProvider: () => KafkaConsumer[Array[Byte], String] = ???
+
+  def producerProvider: () => KafkaProducer[Array[Byte], String] = ???
+
+  class DB {
+    def save(offset: Long, data: String): Future[Done] = ???
+  }
+}
+
+// Consume messages and store a representation, including offset, in DB
+object Example3 extends Example {
+  val db = new DB
+
+  Consumer.source(consumerProvider)
+    .mapAsync(1) { msg =>
+      db.save(msg.record.offset, msg.record.value).map(_ => msg.transaction.commit())
+    }
+}
+
+// Connect a Consumer to Provider
+object Example4a extends Example {
+  Consumer.source(consumerProvider)
+    .map(msg => (new ProducerRecord[Array[Byte], String]("topic2", msg.record.value), msg.transaction))
```

Comment:
> Yes, but eventually (when we unify this kind of API for interop) it might be that the thing that feeds into such a Producer is something like a peristent Actor, or an Http request that will reply only once the thing is committed, etc. So it is the future composability that I am thinking here.

For me as kafka user it is very natural to work in terms of kafka. Why not keep this API kafka related and provide some mid-tier adapters in case of interop API?